src/sources/kubernetes/applicable_transform.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{event::Event, transforms::Transform};
+
+/// Contains several transforms. On the first message, transforms are tried
+/// out one after the other until the first successful one has been found.
+/// After that the transform will always be used.
+///
+/// If nothing succeds the message is still passed.
+pub enum ApplicableTransform {
+    Candidates(Vec<Box<dyn Transform>>),
+    Transform(Option<Box<dyn Transform>>),
+}
+
+impl Transform for ApplicableTransform {
+    fn transform(&mut self, event: Event) -> Option<Event> {
+        match self {
+            Self::Candidates(candidates) => {
+                let candidate = candidates
+                    .iter_mut()
+                    .enumerate()
+                    .find_map(|(i, t)| t.transform(event.clone()).map(|event| (i, event)));
+                if let Some((i, event)) = candidate {
+                    let candidate = candidates.remove(i);
+                    *self = Self::Transform(Some(candidate));
+                    Some(event)
+                } else {
+                    *self = Self::Transform(None);
+                    warn!("No applicable transform.");
```

Comment:
It could happen if user used:
* The newest Kubernetes which had breaking change related to logging, and we haven't yet updated it.
* One of older, pre CRI, Kubernetes version.
* Container runtime with alpha/beta/buggy implementation of CRI.

So, the user could do something about it, if changing either the Kubernetes version or Container runtime is possible. So, something like:
```
Unsupported Kubernetes version and Container runtime pair. Try changing one or the other, and consult with out documentation <link here>.
```

should help.

Reply:
I would like to add these things as comments to the code and link to specs and version numbers.